File: plusOumoins.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

/*
   //main pour la somme et la moyenne  des valeurs d'un tableau
    int somme,tab [3] = {10,5,3};
    //int tableau[4] = {15,81,22,13};
    double moy;
    somme = sommeTableau(tab,3);
    printf("la somme est %d\t\n", somme);
    moy = moyenneTableau(tab,3);
    printf("la moyenne est %.2f\t", moy);


    return 0;
}
*/

int continuerPartie()
{
    int a;
    printf("\nvoulez vous continuer? (0:non) (1:oui)\n");
    scanf("%d", &a);
    return a;
}
void menu ()
{
    printf("\t1 = un nombre entre 1 et 100\n ");
    printf("\t2 = un nombre entre 1 et 1000\n ");
    printf("\t3 = un nombre entre 1 et 10000\n ");
}

void PlusOuMoins(int maxi, int mini)
{
    int a,nombreMystere = 0,cpt = 1;
    //MAX=100,MIN=1;
    srand(time(NULL));
    nombreMystere = (rand()%(maxi - mini + 1)) + mini;
    do
    {

        printf("Donner un nombre svp:\t");
        scanf("%i",&a);

        if (a==nombreMystere)
        {
            printf("Bravo, vous avez trouve le nombre mystere en %d coups\n",cpt);
        }
        else
        {
            if (a>nombreMystere)
            {
                printf("le nombre mystere est inferieur\n");
            }
            else
            {
                printf("le nombre mystere est superieur\n");
            }
        }
        cpt = cpt+1;
    }while (a!=nombreMystere);
}

// calcul de la somme des valeurs d'un tableau
int sommeTableau (int tableau[], int tailleTableau)
{
    int somme =0,i=0;
    for  (i = 0;i<tailleTableau; i++)
    {
        somme = somme + tableau[i];
       // i = i+1;

    }
    return  somme;
}

File: plusOuMoins.py
import random


def continuer_partie():
    print("Voulez vous continuer? (0:non) (1:oui)")
    a = int(input())
    return a


def plus_ou_moins(player):
    """
    L'ordinateur tire au sort un nombre entre 1 et 100.
    Il vous demande de deviner le nombre.
    Vous entrez donc un nombre entre 1 et 100.
    L'ordinateur compare le nombre que vous avez entré avec le nombre « mystère »
    qu'il a tiré au sort. Il vous dit si le nombre mystère est supérieur
    ou inférieur à celui que vous avez entré.
    Puis l'ordinateur vous redemande le nombre. …
    il vous indique si le nombre mystère est supérieur ou inférieur.
    Et ainsi de suite, jusqu'à ce que vous trouviez le nombre mystère.
    :param player: le nom du joueur
    :return:
    """
    print("Tour de ", player)
    nombre_mystere = random.randint(0, 100)  # random integer in range [0, 100]
    cpt = 1     # compteur  qui incrémente à chaque fois que vous passez dans la boucle
    guessing = 1
    print("Donner un nombre svp:\t")
    while guessing == 1:
        guess = int(input())
        if guess == nombre_mystere:
            print(player + " a trouvé le nombre mystere en ", cpt, "coups\n")
            guessing = 0
        else:
            if guess < nombre_mystere:
                print("le nombre mystere est superieur a ", guess)
            else:
                print("le nombre mystere est inferieur a ", guess)
            cpt = cpt+1
    return cpt


def joueur():
    """
    Creer les differents joueurs
    :return:
    """
    name = input("entrer le nom et prenom du joueur SVP: ")
    return name


def jeu():
    """
    Le but du jeu, bien sûr, est de trouver le nombre mystère en un minimum de coups.
    Celui qui ttrouve avec moins de coups gqgne le jeu
    :return:
    """
    joueur1 = joueur()
    joueur2 = joueur()
    compteur1 = plus_ou_moins(joueur1)
    compteur2 = plus_ou_moins(joueur2)

    if compteur1 < compteur2:
        print("*"*55)
        print("Bravo!!!", joueur1, " a gagné en ", compteur1, " coups")
        print("*" * 55)
    else:
        print("*" * 55)
        print("Bravo!!!", joueur2, " a gagné en ", compteur2, " coups")
        print("*" * 55)


jeu()
